### PR TITLE
Clean up PostgreSQL DB creation logic

### PR DESCRIPTION
**Pull request by ekohl**
This started with the goal of getting rid of /root/.pgpass and instead use environment variables, but it escalated. See individual commits for details.

### CODE REVIEW

**Review by bastelfreak — APPROVED** [review]

File: manifests/database/postgresql.pp [hunk 1]
```diff
@@ -23,90 +23,48 @@
...
-    require => File['/root/.pgpass'],
-  }
-
-  file { '/root/.pgpass':
-    ensure => file,
-    mode   => '0600',
-    owner  => 'root',
-    group  => 'root',
-  }
+  $command = "cd ${schema_path} && if [ -f ${schema_file}.gz ]; then zcat ${schema_file}.gz | psql ; else ; psql -f ${schema_file}; fi && touch ${done_file}"
```
Comment (evgeni) [inline manifests/database/postgresql.pp:54]:
```suggestion
  $command = "cd ${schema_path} && if [ -f ${schema_file}.gz ]; then zcat ${schema_file}.gz | psql ; else psql -f ${schema_file}; fi && touch ${done_file}"
```

File: spec/classes/database_postgresql_spec.rb [hunk 2]
```diff
@@ -41,117 +49,75 @@
...
+              super() + [
+                'PGPORT=5432',
+                'PGUSER=zabbix-server',
+                'PGPASSWORD=zabbix-server',
+                'PGDATABASE=zabbix-server',
+              ]
+            end
+
+            it { is_expected.to compile.with_all_deps }
+            it { is_expected.to contain_exec('zabbix_create.sql').with_command("cd #{path} && if [ -f #{sql_server}.gz ]; then zcat #{sql_server}.gz | psql ; else ; psql -f #{sql_server}; fi && touch /etc/zabbix/.schema.done").with_environment(expected_environment) }
```
Comment (evgeni) [inline spec/classes/database_postgresql_spec.rb:74]:
```suggestion
            it { is_expected.to contain_exec('zabbix_create.sql').with_command("cd #{path} && if [ -f #{sql_server}.gz ]; then zcat #{sql_server}.gz | psql ; else psql -f #{sql_server}; fi && touch /etc/zabbix/.schema.done").with_environment(expected_environment) }
```

File: spec/classes/database_postgresql_spec.rb [hunk 3]
```diff
@@ -41,117 +49,75 @@
...
+                'PGUSER=zabbix-proxy',
+                'PGPASSWORD=zabbix-proxy',
+                'PGDATABASE=zabbix-proxy',
+              ]
+            end
+
+            it { is_expected.to compile.with_all_deps }
+
+            if Puppet::Util::Package.versioncmp(zabbix_version, '6.0') < 0
+              it { is_expected.to contain_exec('zabbix_create.sql').with_command("cd #{path} && if [ -f schema.sql.gz ]; then zcat schema.sql.gz | psql ; else ; psql -f schema.sql; fi && touch /etc/zabbix/.schema.done").with_environment(expected_environment) }
```
Comment (evgeni) [inline spec/classes/database_postgresql_spec.rb:107]:
```suggestion
              it { is_expected.to contain_exec('zabbix_create.sql').with_command("cd #{path} && if [ -f schema.sql.gz ]; then zcat schema.sql.gz | psql ; else psql -f schema.sql; fi && touch /etc/zabbix/.schema.done").with_environment(expected_environment) }
```

File: spec/classes/database_postgresql_spec.rb [hunk 4]
```diff
@@ -41,117 +49,75 @@
...
+            else
+              it { is_expected.to contain_exec('zabbix_create.sql').with_command("cd #{path} && if [ -f proxy.sql.gz ]; then zcat proxy.sql.gz | psql ; else ; psql -f proxy.sql; fi && touch /etc/zabbix/.schema.done").with_environment(expected_environment) }
```
Comment (evgeni) [inline spec/classes/database_postgresql_spec.rb:109]:
```suggestion
              it { is_expected.to contain_exec('zabbix_create.sql').with_command("cd #{path} && if [ -f proxy.sql.gz ]; then zcat proxy.sql.gz | psql ; else psql -f proxy.sql; fi && touch /etc/zabbix/.schema.done").with_environment(expected_environment) }
```

**Review by evgeni — APPROVED** [review]

**Comment by evgeni** [comment]
ACK, but not going to merge, @ekohl might or might not want to squash my fixups into the right real commits ;)

**Comment by ekohl** [comment]
I did want to squash those up. I also extended the `Drop unused zabbix_server_{images,data}_sql` commit to do the same thing for MySQL. Made it easier to reason about in the tests :)

**Review by evgeni — APPROVED** [review]

**Review by bastelfreak — APPROVED** [review]

**Pull request merged**